test/codeu/chat/server/DerbyStoreTest.java

```diff
@@ -0,0 +1,190 @@
+package codeu.chat.server;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+
+import codeu.chat.DerbyStore;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+import codeu.chat.common.BasicController;
+import codeu.chat.common.Conversation;
+import codeu.chat.common.Message;
+
+public class DerbyStoreTest {
+	
+	private DerbyStore ds;
+	private Model model;
+	private BasicController controller;
+		
+	@Before
+	public void doBefore() {
+		ds = new DerbyStore();
+		model = new Model();
+	    controller = new Controller(Uuid.NULL, model);
+	}
+	
+	@Test
+	public void checkUserNameExists() {
+		final User user = controller.newUser("user");
+		boolean userNameExists = false;
+		
+		try {
+			userNameExists = ds.checkUsernameExists(user.name);
+		}
+		catch (Exception ex) {
+			
```

Comment:
Empty catch blocks are usually a bad sign. It's not the end of the world here because the test will fail anyway, but it's probably a good idea to at least toss a `printStackTrace()` in here.